src/policy/fees.cpp

```diff
@@ -76,116 +78,109 @@ void TxConfirmStats::UpdateMovingAverages()
     for (unsigned int j = 0; j < buckets.size(); j++)
     {
         for (unsigned int i = 0; i < confAvg.size(); i++)
+        {
             confAvg[i][j] = confAvg[i][j] * decay + curBlockConf[i][j];
+        }
         avg[j] = avg[j] * decay + curBlockVal[j];
-        txCtAvg[j] = txCtAvg[j] * decay + curBlockTxCt[j];
+        txCtAvg[j] = (txCtAvg[j] * decay) + curBlockTxCt[j];
     }
 }
 
 // returns -1 on error conditions
 double TxConfirmStats::EstimateMedianVal(int confTarget,
     double sufficientTxVal,
     double successBreakPoint,
-    bool requireGreater,
     unsigned int nBlockHeight)
 {
+    CAmount minTxFee = 0;
+    // its safe just to parse here because the validity of mintxfee was checked in init.cpp
+
+    // DEFAULT_TRANSACTION_MINFEE = 1000 defined in wallet.h, we hardcode it here as a hack fix
```

Comment:
I think what we really want here is not the wallet's minfee but the network's quasi-consensus min relay fee value?  Our estimated fee should not depend on the wallent in any way, just the network state.  Grab that from chainparams.h DEFAULT_MIN_RELAY_TX_FEE.

I think you hard-coded the 1000 in a few other places too...

Reply:
that is the only place that was hard coded. 

i have switched it to use the minRelayTxFee since this is used to determine the min fee in CreateTransaction